### ISSUE TITLE
[pygame_gui] NotImplementedError on Apple M1

### ISSUE THREAD

**Issue by BUCKFAE** [pygame/pygame#2500]
### Operating system
- macOS Big Sur (11.2.1) - Apple M1

### Python Version
- Python 3.8.2

### Creating a new venv
I create a new venv using PyCharm (same results if I do this via Terminal)

### Python Code
```python
import pygame
import pygame_font

pygame.init()
```

### Installing dependencies
```
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % pip install pygame
Collecting pygame
  Downloading pygame-2.0.1.tar.gz (5.5 MB)
     |████████████████████████████████| 5.5 MB 2.7 MB/s 
Using legacy 'setup.py install' for pygame, since package 'wheel' is not installed.
Installing collected packages: pygame
    Running setup.py install for pygame ... done
Successfully installed pygame-2.0.1
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % pip install pygame_gui
Collecting pygame_gui
  Using cached pygame_gui-0.5.7.tar.gz (1.0 MB)
Requirement already satisfied: pygame>=1.9.5 in ./venv/lib/python3.8/site-packages (from pygame_gui) (2.0.1)
Using legacy 'setup.py install' for pygame-gui, since package 'wheel' is not installed.
Installing collected packages: pygame-gui
    Running setup.py install for pygame-gui ... done
Successfully installed pygame-gui-0.5.7
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % 
```

### Running the program
```
/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/core/interfaces/font_dictionary_interface.py:16: RuntimeWarning: use font: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py)
(ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py))
  bold: bool = False, italic: bool = False) -> pygame.font.Font:
Traceback (most recent call last):
  File "main.py", line 2, in <module>
    import pygame_gui
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/__init__.py", line 7, in <module>
    from pygame_gui.ui_manager import UIManager
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/ui_manager.py", line 5, in <module>
    from pygame_gui.core.interfaces import IUIManagerInterface
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/core/__init__.py", line 1, in <module>
    from pygame_gui.core.ui_appearance_theme import UIAppearanceTheme
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/core/ui_appearance_theme.py", line 14, in <module>
    from pygame_gui.core.interfaces.font_dictionary_interface import IUIFontDictionaryInterface
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/core/interfaces/__init__.py", line 2, in <module>
    from pygame_gui.core.interfaces.font_dictionary_interface import IUIFontDictionaryInterface
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/core/interfaces/font_dictionary_interface.py", line 6, in <module>
    class IUIFontDictionaryInterface(metaclass=ABCMeta):
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame_gui/core/interfaces/font_dictionary_interface.py", line 16, in IUIFontDictionaryInterface
    bold: bool = False, italic: bool = False) -> pygame.font.Font:
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/__init__.py", line 59, in __getattr__
    raise NotImplementedError(missing_msg)
NotImplementedError: font module not available (ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py))
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % 
```

## More information
This error only occurs if I include the import of `pygame_gui`. Pygame itself works flawless on my machine, and I've ensured that I have all the dependencies using
```
brew install sdl2 sdl2_gfx sdl2_image sdl2_mixer sdl2_net sdl2_ttf
```

I've also tried [Mac Compile](https://www.pygame.org/wiki/MacCompile).

Output:
```
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % pip install git+https://github.com/pygame/pygame.git
Collecting git+https://github.com/pygame/pygame.git
  Cloning https://github.com/pygame/pygame.git to /private/var/folders/hk/twd2b0gj6tj32lm6sb264qww0000gn/T/pip-req-build-1wzwksmx
  Running command git clone -q https://github.com/pygame/pygame.git /private/var/folders/hk/twd2b0gj6tj32lm6sb264qww0000gn/T/pip-req-build-1wzwksmx
Using legacy 'setup.py install' for pygame, since package 'wheel' is not installed.
Installing collected packages: pygame
    Running setup.py install for pygame ... done
Successfully installed pygame-2.0.2.dev1
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % 
```
As this does not install pygame_gui I used `pip install pygame_gui` to add that. Running the program results again in the NotImplementedError.

## Test output

```
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % python -m pygame.tests                              
pygame 2.0.2.dev1 (SDL 2.0.14, Python 3.8.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
skipping pygame.tests.cdrom_test (tag 'interactive')
skipping pygame.tests.freetype_test (tag 'ignore')
skipping pygame.tests.ftfont_test (tag 'ignore')
skipping pygame.tests.imageext_test (tag 'ignore')
skipping pygame.tests.mixer_music_test (tag 'ignore')
skipping pygame.tests.mixer_test (tag 'ignore')
skipping pygame.tests.overlay_test (tag 'SDL2_ignore')
skipping pygame.tests.scrap_test (tag 'SDL2_ignore')
skipping pygame.tests.sndarray_test (tag 'ignore')
skipping pygame.tests.surfarray_test (tag 'ignore')
loading pygame.tests.base_test
loading pygame.tests.blit_test
loading pygame.tests.bufferproxy_test
loading pygame.tests.camera_test
loading pygame.tests.color_test
loading pygame.tests.compat_test
loading pygame.tests.constants_test
loading pygame.tests.cursors_test
loading pygame.tests.display_test
loading pygame.tests.draw_test
loading pygame.tests.event_test
loading pygame.tests.fastevent_test
loading pygame.tests.font_test

Stderr:
/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/tests/font_test.py:41: RuntimeWarning: use font: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py)
(ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py))
  pygame_font.init()
loading pygame.tests.gfxdraw_test
loading pygame.tests.image__save_gl_surface_test
loading pygame.tests.image_test
loading pygame.tests.joystick_test
loading pygame.tests.key_test
loading pygame.tests.mask_test
loading pygame.tests.math_test
loading pygame.tests.midi_test
loading pygame.tests.mouse_test
loading pygame.tests.pixelarray_test
loading pygame.tests.pixelcopy_test
loading pygame.tests.rect_test
loading pygame.tests.rwobject_test
loading pygame.tests.sprite_test
loading pygame.tests.surface_test
loading pygame.tests.surflock_test
loading pygame.tests.sysfont_test
loading pygame.tests.threads_test
loading pygame.tests.time_test
loading pygame.tests.touch_test
loading pygame.tests.transform_test
loading pygame.tests.version_test
loading pygame.tests.video_test
Traceback (most recent call last):
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 193, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/tests/__main__.py", line 144, in <module>
    run_and_exit(*args, **kwds)
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/tests/test_utils/run_tests.py", line 352, in run_and_exit
    total, fails = run(*args, **kwargs)
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/tests/test_utils/run_tests.py", line 309, in run
    raise AssertionError(
AssertionError: Something went wrong in the Test Machinery:
total: 1484 != untrusty_total: 1483
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck
```

**Comment by MyreMylar** [comment]
Looks like the font module doesn't work on M1 Mac, judging from the output of the tests and the reported errors. I'm guessing `pygame_gui `is triggering the error because it actually tries to create a Font object. Perhaps there is an issue with the underlying freetype library on the new macs.

**Comment by BUCKFAE** [comment]
> Looks like the font module doesn't work on M1 Mac, judging from the output of the tests and the reported errors. I'm guessing `pygame_gui `is triggering the error because it actually tries to create a Font object. Perhaps there is an issue with the underlying freetype library on the new macs.

Can confirm this:

```import pygame
# import pygame_gui

pygame.init()
pygame.font.init()
```


Stil raises this exception:

```
pygame 2.0.2.dev1 (SDL 2.0.14, Python 3.8.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
/Users/julianschubert/PycharmProjects/DependencyCheck/main.py:5: RuntimeWarning: use font: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py)
(ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py))
  pygame.font.init()
Traceback (most recent call last):
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/main.py", line 5, in <module>
    pygame.font.init()
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/__init__.py", line 59, in __getattr__
    raise NotImplementedError(missing_msg)
NotImplementedError: font module not available (ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv/lib/python3.8/site-packages/pygame/font.py))

Process finished with exit code 1
```

**Comment by danielleontiev** [comment]
I had the same issue on MBP 16 (2019) with MacOS BigSur even it has Intel chip.

I've tried to do some sort of monkey patching and built pygame on Mac and on Linux and noticed that the package on Mac has extra file `font.py` in output artifact that I did not see on Linux. This `font.py` contains import from `pygame.sysfont` and `sysfont.py` has import from `pygame.font` - looks like cyclic import, but I am not sure due to lack of expertise in Python.

I do not see this file on Linux and I suspect this line https://github.com/pygame/pygame/blob/main/setup.py#L362 causes the issue.

Next I've run `python3 -m buildconfig -auto` and found that `font` module is not available. I installed sdl2 font module with `brew install SDL2_ttf` and after that installed pygame with `pip install git+https://github.com/pygame/pygame`. There is no more `font.py` that causes cyclic import and I've finally run game sucessfully

But maybe it's just coinsedence

**Comment by BUCKFAE** [comment]
I had sdl2_ttf already installed on my machine, nevertheless I used `brew reinstall sdl2_ttf` and rebooted the laptop. I even created a new venv (just to make sure)

In the new env I installed pygame:
```
(venv2) (base) julianschubert@MBP-von-Julian DependencyCheck % pip install pygame
Collecting pygame
  Using cached pygame-2.0.1.tar.gz (5.5 MB)
Using legacy 'setup.py install' for pygame, since package 'wheel' is not installed.
Installing collected packages: pygame
    Running setup.py install for pygame ... done
Successfully installed pygame-2.0.1
(venv2) (base) julianschubert@MBP-von-Julian DependencyCheck % 
```

Executed my (unchanged) `main.py` using `python3 main.py`.
Error message:
```
pygame 2.0.1 (SDL 2.0.14, Python 3.8.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
main.py:5: RuntimeWarning: use font: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/font.py)
(ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/font.py))
  pygame.font.init()
Traceback (most recent call last):
  File "main.py", line 5, in <module>
    pygame.font.init()
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/__init__.py", line 59, in __getattr__
    raise NotImplementedError(missing_msg)
NotImplementedError: font module not available (ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/font.py))
(venv2) (base) julianschubert@MBP-von-Julian DependencyCheck % 
```

Running `python3 -m main.py -auto` yields the following message:

```
(venv2) (base) julianschubert@MBP-von-Julian DependencyCheck % python3 -m main.py -auto
pygame 2.0.1 (SDL 2.0.14, Python 3.8.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
/Users/julianschubert/PycharmProjects/DependencyCheck/main.py:5: RuntimeWarning: use font: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/font.py)
(ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/font.py))
  pygame.font.init()
Traceback (most recent call last):
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 184, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/main.py", line 5, in <module>
    pygame.font.init()
  File "/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/__init__.py", line 59, in __getattr__
    raise NotImplementedError(missing_msg)
NotImplementedError: font module not available (ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' (most likely due to a circular import) (/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame/font.py))
(venv2) (base) julianschubert@MBP-von-Julian DependencyCheck % 
```

Regarding the file `font.py`:
I created a new venv as described above on a Mac, a Windows machine and a Linux machine.

<details open>
<summary>MacOS</summary>

```bash
(base) julianschubert@MBP-von-Julian pygame % pwd
/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame
(base) julianschubert@MBP-von-Julian pygame % ls *font*
font.py		font.pyi	ftfont.py	sysfont.py
(base) julianschubert@MBP-von-Julian pygame % 
```
</details>

<details open>
<summary>WSL</summary>

```bash
❯ ls lib/python3.9/site-packages/pygame/*font*
lib/python3.9/site-packages/pygame/font.cpython-39-x86_64-linux-gnu.so
lib/python3.9/site-packages/pygame/font.pyi
lib/python3.9/site-packages/pygame/ftfont.py
lib/python3.9/site-packages/pygame/sysfont.py
```
</details>


<details open>
<summary>Windows</summary>

```bash
(windows) PS D:\test-venv\windows> ls .\Lib\site-packages\pygame\*font*


    Verzeichnis: D:\test-venv\windows\Lib\site-packages\pygame


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       04.03.2021     09:15          19968 font.cp37-win32.pyd
-a----       04.03.2021     09:15           1771 font.pyi
-a----       04.03.2021     09:15           6284 ftfont.py
-a----       04.03.2021     09:15          15583 sysfont.py
```
</details>

Content of the file `font.py` (on macOS)
```python
"""pygame module for loading and rendering fonts (freetype alternative)"""

__all__ = ['Font', 'init', 'quit', 'get_default_font', 'get_init', 'SysFont',
           "match_font", "get_fonts"]

from pygame._freetype import init, Font as _Font, get_default_resolution
from pygame._freetype import quit, get_default_font, get_init as _get_init
from pygame._freetype import __PYGAMEinit__
from pygame.sysfont import match_font, get_fonts, SysFont as _SysFont
from pygame import encode_file_path
from pygame.compat import bytes_, unicode_, as_unicode, as_bytes


class Font(_Font):
    """Font(filename, size) -> Font
       Font(object, size) -> Font
       create a new Font object from a file (freetype alternative)

       This Font type differs from font.Font in that it can render glyphs
       for Unicode code points in the supplementary planes (> 0xFFFF).
       """

    __encode_file_path = staticmethod(encode_file_path)
    __get_default_resolution = staticmethod(get_default_resolution)
    __default_font = encode_file_path(get_default_font())

    __unull = as_unicode(r"\x00")
    __bnull = as_bytes("\x00")

    def __init__(self, file, size=-1):
        if size <= 1:
            size = 1
        if isinstance(file, unicode_):
            try:
                bfile = self.__encode_file_path(file, ValueError)
            except ValueError:
                bfile = ''
        else:
            bfile = file
        if isinstance(bfile, bytes_) and bfile == self.__default_font:
            file = None
        if file is None:
            resolution = int(self.__get_default_resolution() * 0.6875)
            if resolution == 0:
                resolution = 1
        else:
            resolution = 0
        super(Font, self).__init__(file, size=size, resolution=resolution)
        self.strength = 1.0 / 12.0
        self.kerning = False
        self.origin = True
        self.pad = True
        self.ucs4 = True
        self.underline_adjustment = 1.0

    def render(self, text, antialias, color, background=None):
        """render(text, antialias, color, background=None) -> Surface
           draw text on a new Surface"""

        if text is None:
            text = ""
        if (isinstance(text, unicode_) and self.__unull in text):
            raise ValueError("A null character was found in the text")
        if (isinstance(text, bytes_) and self.__bnull in text):
            raise ValueError("A null character was found in the text")
        save_antialiased = self.antialiased
        self.antialiased = bool(antialias)
        try:
            s, r = super(Font, self).render(text, color, background)
            return s
        finally:
            self.antialiased = save_antialiased

    def set_bold(self, value):
        """set_bold(bool) -> None
           enable fake rendering of bold text"""

        self.wide = bool(value)

    def get_bold(self):
        """get_bold() -> bool
           check if text will be rendered bold"""

        return self.wide

    bold = property(get_bold, set_bold)

    def set_italic(self, value):
        """set_italic(bool) -> None
           enable fake rendering of italic text"""

        self.oblique = bool(value)

    def get_italic(self):
        """get_italic() -> bool
           check if the text will be rendered italic"""

        return self.oblique

    italic = property(get_italic, set_italic)

    def set_underline(self, value):
        """set_underline(bool) -> None
           control if text is rendered with an underline"""

        self.underline = bool(value)

    def get_underline(self):
        """set_bold(bool) -> None
           enable fake rendering of bold text"""

        return self.underline

    def metrics(self, text):
        """metrics(text) -> list
           Gets the metrics for each character in the passed string."""

        return self.get_metrics(text)

    def get_ascent(self):
        """get_ascent() -> int
           get the ascent of the font"""

        return self.get_sized_ascender()

    def get_descent(self):
        """get_descent() -> int
           get the descent of the font"""

        return self.get_sized_descender()

    def get_height(self):
        """get_height() -> int
           get the height of the font"""

        return self.get_sized_ascender() - self.get_sized_descender() + 1

    def get_linesize(self):
        """get_linesize() -> int
           get the line space of the font text"""

        return self.get_sized_height()

    def size(self, text):
        """size(text) -> (width, height)
           determine the amount of space needed to render text"""

        return self.get_rect(text).size

FontType = Font

def get_init():
    """get_init() -> bool
       true if the font module is initialized"""

    return _get_init()

def SysFont(name, size, bold=0, italic=0, constructor=None):
    """pygame.ftfont.SysFont(name, size, bold=False, italic=False, constructor=None) -> Font
       Create a pygame Font from system font resources.

       This will search the system fonts for the given font
       name. You can also enable bold or italic styles, and
       the appropriate system font will be selected if available.

       This will always return a valid Font object, and will
       fallback on the builtin pygame font if the given font
       is not found.

       Name can also be an iterable of font names, a string of
       comma-separated font names, or a bytes of comma-separated
       font names, in which case the set of names will be searched
       in order. Pygame uses a small set of common font aliases. If the
       specific font you ask for is not available, a reasonable
       alternative may be used.

       If optional constructor is provided, it must be a function with
       signature constructor(fontpath, size, bold, italic) which returns
       a Font instance. If None, a pygame.ftfont.Font object is created.
    """
    if constructor is None:
        def constructor(fontpath, size, bold, italic):
            font = Font(fontpath, size)
            font.set_bold(bold)
            font.set_italic(italic)
            return font

    return _SysFont(name, size, bold, italic, constructor)

del _Font, get_default_resolution, encode_file_path, as_unicode, as_bytes
```

I am also not that experienced with python, but your assumption that the problem is in fact a cyclic import.

**Comment by nelsonlove** [comment]
I've found what appears to be a simple fix for this issue by moving the SysFont import in ftfont.py to immediately before the SysFont function on line 158. See my fork at https://github.com/nelsonlove/pygame. Hope this helps!

**Comment by illume** [comment]
Is someone able to test the PR at https://github.com/pygame/pygame/pull/2508 ? Or maybe explain how it fixes it?

I don't understand the fix, and can't reproduce it myself.

**Comment by llindstrom** [comment]
I have reproduced the problem on linux and traced it to this commit: d7b88de8176eea33987a9146d36b1251980da88e. It has nothing to do with venv or pygame_gui.

Look at the **Test output** for the `python -m pygame.tests` unit test run posted above:

```
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % python -m pygame.tests                      
pygame 2.0.2.dev1 (SDL 2.0.14, Python 3.8.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
skipping pygame.tests.cdrom_test (tag 'interactive')
skipping pygame.tests.freetype_test (tag 'ignore')
skipping pygame.tests.ftfont_test (tag 'ignore')
skipping pygame.tests.imageext_test (tag 'ignore')
skipping pygame.tests.mixer_music_test (tag 'ignore')
skipping pygame.tests.mixer_test (tag 'ignore')
skipping pygame.tests.overlay_test (tag 'SDL2_ignore')
skipping pygame.tests.scrap_test (tag 'SDL2_ignore')
skipping pygame.tests.sndarray_test (tag 'ignore')
skipping pygame.tests.surfarray_test (tag 'ignore')
...
```

Notice all the skipped tests. The extension modules for SDL2_image and SDL_mixer where not built.

pygame for macOS is built and installed from source.

```
(venv) (base) julianschubert@MBP-von-Julian DependencyCheck % pip install pygame
Collecting pygame
  Downloading pygame-2.0.1.tar.gz (5.5 MB)
     |████████████████████████████████| 5.5 MB 2.7 MB/s 
Using legacy 'setup.py install' for pygame, since package 'wheel' is not installed.
Installing collected packages: pygame
    Running setup.py install for pygame ... done
Successfully installed pygame-2.0.1
```

For the build process to work, all the dependencies, such as SDL2, SDL2_image, SDL_mixer and SDL2_ttf must be installed first. Otherwise the corresponding pygame extension modules will not be built and installed. SDL2 was found as the build did not failed. The other libraries are considered optional.

The `pygame.font` unit tests ran because the test framework assumes the font module is present (This can be corrected in pygame/tests/font_tags.py.) It failed with the ImportError exception:

```
loading pygame.tests.font_test

Stderr:
...
(ImportError: cannot import name 'Font' from partially initialized module 'pygame.font' 
    (most likely due to a circular import) (... lib/python3.8/site-packages/pygame/font.py))
  pygame_font.init()
```

When pygame is built without the SDL2_ttf library present, it assumes that only the `pygame.freetype` library will be used. For backwards compatibility `pygame.ftfont` is installed as a replacement for the SDL2_ttf based extension module. This is the _font.py_ file in the above error message. It is also seen in this directory listing:

```
(base) julianschubert@MBP-von-Julian pygame % pwd
/Users/julianschubert/PycharmProjects/DependencyCheck/venv2/lib/python3.8/site-packages/pygame
(base) julianschubert@MBP-von-Julian pygame % ls *font*
font.py		font.pyi	ftfont.py	sysfont.py
(base) julianschubert@MBP-von-Julian pygame % 
```

but not here for Windows, where the SDL2_ttf extension library was installed:

```
(windows) PS D:\test-venv\windows> ls .\Lib\site-packages\pygame\*font*


    Verzeichnis: D:\test-venv\windows\Lib\site-packages\pygame


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       04.03.2021     09:15          19968 font.cp37-win32.pyd
-a----       04.03.2021     09:15           1771 font.pyi
-a----       04.03.2021     09:15           6284 ftfont.py
-a----       04.03.2021     09:15          15583 sysfont.py
```

Anyway, `pygame.ftfont` should not import `pygame.font`. However, it does import `pygame.sysfont`. For this reason the `pygame.sysfont` module deliberately avoided importing `pygame.font` at initialization. Instead it postponed importing `pygame.font` until needed using a function call. This avoided the circular import.

Apparently importing a module inside a function upsets pylint, so the `pygame.font` import was moved to the top of the file in the commit "[Corrects several pylint warnings in sysfont module](d7b88de8176eea33987a9146d36b1251980da88e)". This reintroduces the circular import and causes the confusing ImportError message. I do not see how the pull request [Update ftfont.py to fix circular dependency](https://github.com/pygame/pygame/pull/2508) fixes the problem.

**Comment by nelsonlove** [comment]
Some additional information: I installed using Homebrew in the same manner as the poster above. The issue appeared for me without pygame_gui being present, and I also noticed the tests being skipped during installation. The font module is working now but the extended image module does not appear to be installed.

I also tried running some of the example demos included in the py-sdl2 library, which produced "mach-o, but wrong architecture" warnings for sdl2_image and sdl2_ttf. It looked like font and image rendering were being skipped in the demos but they appeared to work otherwise.

Apologies if the pull request was premature—I'm new on GitHub and admittedly in over my head here—but I wanted to share something that worked for me.

**Comment by nelsonlove** [comment]
For some reason the SDL2 libraries aren't being seen when installed by Homebrew. I traced the issue back to `buildconfig/config_darwin.py`, which looks for needed files in `/usr/local` instead of `/opt/homebrew`. Additionally, the symlinks created by Homebrew from, i.e., `/opt/homebrew/Cellar/sdl2_image/2.0.5/include` to `opt/homebrew/include` didn't seem to be getting followed when I changed mentions of `usr/local` to `opt/homebrew`.

My [fork](https://github.com/nelsonlove/pygame) now ensures symlinks aren't being passed by buildconfig/config_darwin.py. Pygame now sees all SDL2 modules on build for me.

**Comment by Muxelmann** [comment]
It's been a while, but it appears that by including the new Homebrew paths into the `config_darwin.py` fixes the issue for Font, Sound (i.e. missing/not found modules/libraries) and Image (only being able to load Windows BMP).

Hence, I took the liberty and created a pull request with only that change: https://github.com/pygame/pygame/pull/2636

When installing this branch I don't see a need to touch the `font.py` file. After all, everything seems to work fine now, given that `sdl2` etc. is installed using Homebrew.

**Comment by Starbuck5** [comment]
I believe this is fixed by #2636
